Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -412,6 +412,92 @@ public override int GetHashCode()
             }
         }
 
+        /// <summary>
+        /// Helper class for managing the visibility of the gaze pointer to match windows mixed reality and HoloLens 2
+        /// When application starts, gaze pointer is visible. Then when articulate hands / motion controllers
+        /// appear, hide the gaze cursor. Whenever user says the voice wake word, make the 
+        /// gaze controller appear.
+        /// </summary>
+        private class GazePointerStateMachine
```

Comment:
Code looks good to me, only comment here is that this code looks super testable (i.e. it takes deterministic inputs, and then its state changes) - this would be a great candidate for a unit test.

Reply:
Just added some playmode tests to validate the state of cursor giving hand states.